packages/frontend/src/api.tsx

```diff
@@ -64,10 +107,21 @@ async function betterFetch<T>(
 
   const responsePayload = await response.json()
   console.log('HTTP response: ', responsePayload)
+  if (responsePayload.field) {
+    throw new FieldError(responsePayload.error, responsePayload.field)
+  }
 
   if (response.status < 200 || response.status > 299) {
     throw new Error(`${response.status} ${response.statusText}`)
   }
 
   return responsePayload
 }
+
+export class FieldError extends Error {
```

Comment:
Is it safe to extend the JS Error directly? Shouldn't we use something like https://www.npmjs.com/package/make-error?